lang/java/avro/src/main/java/org/apache/avro/Protocol.java

```diff
@@ -390,13 +391,14 @@ public boolean equals(Object o) {
     if (!(o instanceof Protocol))
       return false;
     Protocol that = (Protocol) o;
-    return this.name.equals(that.name) && this.namespace.equals(that.namespace) && this.types.equals(that.types)
-        && this.messages.equals(that.messages) && this.propsEqual(that);
+    return Objects.equals(this.name, that.name) && Objects.equals(this.namespace, that.namespace)
+        && Objects.equals(this.types, that.types) && Objects.equals(this.messages, that.messages)
+        && this.propsEqual(that);
   }
 
   @Override
   public int hashCode() {
-    return name.hashCode() + namespace.hashCode() + types.hashCode() + messages.hashCode() + propsHashCode();
+    return name.hashCode() + Objects.hash(namespace) + types.hashCode() + messages.hashCode() + propsHashCode();
```

Comment:
Why not add the other parameters to `Objects.hash`? That would read even better.

Reply:
New hash would be backward incompatible with the old one in that case. If that's ok, I do not see any problems switching to `Objects.hash`

Reply:
@opwvhk So what do you think?

Reply:
Backwards compatibility is not an issue.

The `Protocol#hashCode()` method has never specified differently, and thus uses the general contract of `hashCode()`. That states that multiple invocations on the same object, if unchanged, yield the same result. However, it is explicitly stated that the result "need not remain consistent from one execution of an application to another execution of the same application".

See https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Object.html#hashCode()

Reply:
Thanks! I've updated `Protocol#hashCode`